Nucleo f030r8

MaciejDebski opened the pull request:
ZEP-2492
boards: provide support for Nucleo-64 F030R8

Commit:
arch: arm: soc: provide support for stm32f0.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
dts: arm: provide support for nucleo f030r8.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
boards: arm: provide support for nucleo f030r8.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
drivers: clock_control: provide support for stm32f0.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
drivers: gpio: provide support for stm32f0.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
drivers: interrupt_controller: provide support for stm32f0.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
drivers: pinmux: provide support for stm32f0 and nucleo f030r8 board.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
drivers: serial: provide support for stm32f0.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

MaciejDebski commented:
@apodogrocki, @erwango,  @ydamigos
Would you be so kind as to review my code, please?

ydamigos reviewed (changes requested):
After the introduction of pinctrl nodes and uart pinctrl configuration in stm32 socs DT files, the F0 family should follow the same approach. Please consider to add also a stm32f0.dtsi file for the F0 series.

dbkinder commented on boards/arm/nucleo_f030r8/doc/nucleof334r8.rst:
```diff
@@ -0,0 +1,181 @@
+
+Overview
+********
+STM32 Nucleo-64 development board with STM32F030R8 MCU, supports Arduino and ST morpho connectivity.
```
The STM32 ...

dbkinder commented on boards/arm/nucleo_f030r8/doc/nucleof334r8.rst:
```diff
@@ -0,0 +1,181 @@
+
+The STM32 Nucleo board provides an affordable and flexible way for users to try out new concepts,
+and build prototypes with the STM32 microcontroller, choosing from the various
+combinations of performance, power consumption and features.
```
add a comma before the "and"

dbkinder commented on boards/arm/nucleo_f030r8/doc/nucleof334r8.rst:
```diff
@@ -0,0 +1,181 @@
+expansion of the STM32 Nucleo open development platform with a wide choice of
+specialized shields.
+
+The STM32 Nucleo board does not require any separate probe as it integrates the ST-LINK/V2-1
```
recommend deleting "does not require any separate probe as it"

dbkinder commented on boards/arm/nucleo_f030r8/doc/nucleof334r8.rst:
```diff
@@ -0,0 +1,181 @@
+========
+
+Nucleo F030R8 board includes an ST-LINK/V2-1 embedded debug tool interface.
+This interface is supported by the openocd version included in Zephyr SDK.
```
in the Zephyr SDK.

dbkinder commented on boards/arm/nucleo_f030r8/doc/nucleof334r8.rst:
```diff
@@ -0,0 +1,181 @@
+Flashing an application to Nucleo F030R8
+----------------------------------------
+
+The sample application `blinky` is being used in this tutorial:
```
change to

     :ref:`blinky-sample`

dbkinder commented on boards/arm/nucleo_f030r8/doc/nucleof334r8.rst:
```diff
@@ -0,0 +1,181 @@
+----------------------------------------
+
+The sample application `blinky` is being used in this tutorial:
+
```
I'd recommend deleting this code block (and change the colon ":" on the previous line to end with a period.

dbkinder commented on boards/arm/nucleo_f030r8/doc/nucleof334r8.rst:
```diff
@@ -0,0 +1,181 @@
+
+   $ make BOARD=nucleo_f030r8 flash
+
+Run a serial host program to connect with your Nucleo board:
```
Just checking, do you need to connect a serial host for this app?  What's written to the console?

dbkinder reviewed (changes requested):
a few recommended changes

galak reviewed (changes requested):
The hal import should separate the import of code from the adding Kconfig/Makefile for building it.

Also, the commit message should include the source (where to find the HAL)

Example:

commit 00155448cf9bee26eef8a9fec23a9e7c9e92c90f
Author: Erwan Gouriou <erwan.gouriou@linaro.org>
Date:   Wed Nov 9 15:23:39 2016 +0100

    stm32cube: Introduce STM32Cube for STM32F2xx series
    
    With this commit we provide header and HAL/LL files for all STM32F2
    SoCs.
    Using STM32Cube provides generic access to all families, maximizes
    code reuse across different STM32 families and helps taking benefit
    of a mature SDK
    
    Origin: STM32Cube
    URL: http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef2.html
    Purpose: Introduction of STM32F2xx STM32Cube definitions
    and Abstraction Layers
    Maintained-by: External
    
    Original STM32Cube tree structure has been modified to a minimum
    structure for a better fit into Zephyr. hal is split into 2 parts:
    -driver: initially Drivers/STM32F2xx_HAL_Driver, contains HAL and LL
    -soc: initially Drivers/CMSIS/Device/ST/STM32F2xx, contains stm32f2xx
    socs header files
    
    Change-Id: I78a4716d495ce3f3038a442e19a0403f0f00506c
    Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

nashif commented:
> Also, the commit message should include the source (where to find the HAL) 

please look at CONTRIBUTING.rst or https://www.zephyrproject.org/doc/contribute/contribute_guidelines.html#identifying-contribution-origin

galak commented on drivers/interrupt_controller/Kconfig.stm32:
```diff
@@ -13,6 +13,36 @@ config EXTI_STM32
 	help
 	 Enable EXTI driver for STM32 line of MCUs
 
+if SOC_SERIES_STM32F0X
```
why do we need this if?  We dont for any other of the STM32 SoC families

galak commented on drivers/interrupt_controller/exti_stm32.c:
```diff
@@ -270,7 +304,6 @@ static inline void __stm32_exti_isr_15_10(void *arg)
 	__stm32_exti_isr(10, 16, arg);
 }
 
-#ifdef CONFIG_SOC_SERIES_STM32F4X
```
seems like this shouldn't be removed

Seems like #if defined(CONFIG_SOC_SERIES_STM32F0X) || defined(CONFIG_SOC_SERIES_STM32F4X)

galak reviewed (changes requested):
Some changes/issues with the interrupt changes

Commit:
ext: hal: stm32cube: HAL/LL static library for stm32f0xx v.1.9.

The commit delivers HAL/LL static library source files for STM32F0 SoC.
Using STM32Cube provides generic access to all families, maximizes
code reuse across different STM32 families and helps taking benefits
of a mature SDK.

Origin: STM32Cube
URL: http://www.st.com/en/embedded-software/stm32cubef0.html
Purpose: Introduction of STM32F0xx STM32Cube definitions
and Abstraction Layers
Maintained-by: External

Original STM32Cube tree structure has been modified to a minimum
structure for a better fit into Zephyr. HAL is split into 2 parts:
-driver: initially Drivers/STM32F0xx_HAL_Driver, contains HAL and LL
-soc: initially Drivers/CMSIS/Device/ST/STM32F0xx, contains stm32f0xx
SoCs header files.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

Commit:
ext: hal: stm32cube: provide build method for stm32f0 HAL/LL lib.

JIRA: ZEP-2492

Signed-off-by: Maciej Debski <maciej.debski@rndity.com>

MaciejDebski commented on drivers/pinmux/Makefile:
```diff
@@ -6,20 +6,21 @@ ccflags-$(CONFIG_PINMUX_QMSI) += -I$(CONFIG_QMSI_INSTALL_PATH)/include
 obj-$(CONFIG_PINMUX_MCUX) += pinmux_mcux.o
 obj-$(CONFIG_PINMUX_STM32) += stm32/pinmux_stm32.o
 obj-$(CONFIG_PINMUX_BEETLE) += beetle/pinmux_board_v2m_beetle.o
-obj-$(CONFIG_BOARD_NUCLEO_F103RB) += stm32/pinmux_board_nucleo_f103rb.o
```
The changes are just to put the nucleo boards in order.
As for pinmux, I will follow the include/dt-bindings convention, and I will leave the boards/ as it is in the rest of nucleos.

galak commented:
@MaciejDebski is this now working for you with the pinmux/dts changes?

MaciejDebski commented:
@galak Yes, I did manage to make it work for new pinmux/dts in the end, but there were a problem on the way, totally somewhere else. Some small snippet of code concerning aliasing flash memory in my cortex, which was added recently by someone else, was messing everything up. I described it in this months zephyr-devel, in the mail named: Adding Nucleo-F030R8 support to Zephyr - runtime error.

I still have to post issue on github, but I am not going to change my pull request code, it should be all good. Is it OK?

MaciejDebski commented:
@nashif Thank you for your help, it is fixed now!

MaciejDebski commented:
@galak HAL libs and Kconfig/Makefile are now separated. Thank you.

MaciejDebski commented:
@dbkinder All the recommended changes are implemented, thank you!

MaciejDebski commented:
@ydamigos The new approach to pinmux/dts is implemented, thank you!

ydamigos commented on arch/arm/soc/st_stm32/stm32f0/soc.h:
```diff
@@ -0,0 +1,63 @@
+#include <stm32f0xx.h>
+
+/* IO pin functions */
+enum stm32f0x_pin_config_mode {
```
Do we still need this enumeration?

erwango commented on drivers/clock_control/stm32_ll_clock.c:
```diff
@@ -223,7 +236,9 @@ static int stm32_clock_control_init(struct device *dev)
 
 	/* Disable other clocks */
 	LL_RCC_HSE_Disable();
+#ifndef CONFIG_SOC_SERIES_STM32F0X
```
Remove the #ifdef here, as you defined LL_RCC_MSI_Disable in stm32_f0_ll_clock.c file
It helps reducing the use of ifdef in this generic file (when doable)

erwango commented on drivers/clock_control/stm32_ll_clock.c:
```diff
@@ -239,7 +254,10 @@ static int stm32_clock_control_init(struct device *dev)
 	/* Disable other clocks */
 	LL_RCC_HSI_Disable();
+
+#ifndef CONFIG_SOC_SERIES_STM32F0X
```
Same here

erwango commented on drivers/clock_control/stm32_ll_clock.c:
```diff
@@ -274,15 +292,19 @@ static int stm32_clock_control_init(struct device *dev)
 	LL_RCC_HSI_Disable();
+#ifndef CONFIG_SOC_SERIES_STM32F0X
 	LL_RCC_MSI_Disable();
+#endif /* CONFIG_SOC_SERIES_STM32F0X */
```
same here

erwango commented on drivers/clock_control/stm32f0x_ll_clock.c:
```diff
@@ -0,0 +1,81 @@
+					<< RCC_CFGR_PLLMUL_Pos);
+#if defined(RCC_PLLSRC_PREDIV1_SUPPORT)
+	/* PREDIV support is a specific RCC configuration present on */
+	/* following SoCs: STM32F302XE, STM32F303xE and STM32F398xx */
```
Update comment

MaciejDebski commented:
@ydamigos enum in arch/arm/soc/st_stm32/stm32f0/soc.h was not necessary to compile. Updated. Thank you.

erwango commented on drivers/pinmux/Makefile:
```diff
@@ -6,20 +6,21 @@ ccflags-$(CONFIG_PINMUX_QMSI) += -I$(CONFIG_QMSI_INSTALL_PATH)/include
 obj-$(CONFIG_PINMUX_MCUX) += pinmux_mcux.o
 obj-$(CONFIG_PINMUX_STM32) += stm32/pinmux_stm32.o
 obj-$(CONFIG_PINMUX_BEETLE) += beetle/pinmux_board_v2m_beetle.o
-obj-$(CONFIG_BOARD_NUCLEO_F103RB) += stm32/pinmux_board_nucleo_f103rb.o
```
STM32 boards pinmux code has been moved recenlty with #1301 (merged in ARM branch for now).
Please update according to this change (maybe rebase on ARM branch to do so)

MaciejDebski replied:
@erwango I moved the drivers/pinmux/stm32/pinmux_board_nucleo_f030r8.c to boards/arm/nucleo_f030r8/pinmux.c. Is that what you mean?

erwango replied:
Yes, and then you should not need to impact pinmux makefile anymore

MaciejDebski commented:
@erwango I did changes to to drivers/clock_control/ as you suggested, thank you!

superna9999 reviewed (commented):
Hi,

Could you bump the version of the STM32Cube to the latest 1.9.0 release ?

Thanks,
Neil

MaciejDebski commented:
@superna9999 
Updated to 1.9, thank you!

b0661 committed:
arch: arm: core: fix vector table relocate write to flash

Some SOCs (e.g. STM32F0) can map the flash to address 0 and
the flash base address at the same time. Prevent writing to
duplicate flash address which stops the SOC.

Allow Cortex M SOCs to create their own vector table relocation
function.

Provide a relocation function for STM32F0x SOCs.

Signed-off-by: Bobby Noelte <b0661n0e17e@gmail.com>

superna9999 commented:
Hi @MaciejDebski @galak 

the issue about missing memory was discusses on the mailing-list, it should be solved since all other points were solved.

b0661 commented:
Hello @MaciejDebski @superna9999

the commit [https://github.com/b0661/zephyr/commit/30e4e7142aad67b097338ff34c83ad15cb83a5c0r ](https://github.com/b0661/zephyr/commit/30e4e7142aad67b097338ff34c83ad15cb83a5c0r) should make the pull request pass CI.

@superna9999 You mentioned that the missing memory was already discussed on the mailing list and solved. I did not find the solution. Don´t we need the commit above?

@MaciejDebski Do you plan to integrate the commit? If you like I can create a combined pull request that takes your pull request as the basis [https://github.com/b0661/zephyr/tree/stm32f0_nucleos](https://github.com/b0661/zephyr/tree/stm32f0_nucleos)?

Regards
Bobby

superna9999 commented:
@b0661 In was referencing the solution you provide in this commit

superna9999 commented:
@MaciejDebski Would you want @b0661 or me to take over this PR (while keeping your authorship) if you don't have enough time ?
We have at least 2 more F0 SoCs/Boards to submit following this PR

apodogrocki commented:
Hi Neil,

On behalf of Maciek I would like to ask you or Bobby to take over this PR.
In October Maciek will not have enough time to finalize this job.

Cheers,
Adam Podogrocki

2017-10-09 13:55 GMT+02:00 Neil Armstrong <notifications@github.com>:

> @MaciejDebski <https://github.com/maciejdebski> Would you want @b0661
> <https://github.com/b0661> or me to take over this PR (while keeping your
> authorship) if you don't have enough time ?
> We have at least 2 more F0 SoCs/Boards to submit following this PR
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-335136071>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP5E9W_K3-G1V6SwLBWIKCNqnPORSuZ4ks5sqgmngaJpZM4O2FiW>
> .
>



-- 

Adam Podogrocki

*Senior Software Engineer* of *rndity;*

+48 607 236 982 <+48+607+236+982>  rndity.com

[image: facebook] <http://facebook.com/rndity> [image: twitter]
<http://twitter.com/RnDity_com> [image: linkedin]
<https://www.linkedin.com/company/rndity> [image: github]
<https://github.com/RnDity>

[image: rndity]

Niniejsza wiadomość wraz z załącznikami może zawierać chronione prawem
lub poufne informacje i została wysłana wyłącznie do wiadomości i
użytku osób, do których została zaadresowana. Jeśli wiadomość została
otrzymana przypadkowo zabrania się jej kopiowania lub rozsyłania do
osób trzecich. W takim przypadku uprasza się o natychmiastowe
zniszczenie wiadomości oraz poinformowanie nadawcy o zaistniałej
sytuacji za pomocą wiadomości zwrotnej. Dziękujemy.

This message, including any attachments hereto, may contain privileged
or confidential information and is sent solely for the attention and
use of the intended addressee(s). If you are not an intended
addressee, you may neither use this message nor copy or deliver it to
anyone. In such case, you should immediately destroy this message and
kindly notify the sender by reply email. Thank you.

erwango commented:
Hi Adam,

Should we consider this is same situation for you and
https://github.com/zephyrproject-rtos/zephyr/pull/556 ?
("iwdg: make independent watchdog driver compliant with STM32Cube LL API")
(keeping your credit of course).
There is less demand on this one than on STM32F0, but it would be nice we
don't loose the benefit of your work.
Or you already planned to resume it later on?

Cheers
Erwan

On 9 October 2017 at 14:13, Adam Podogrocki <notifications@github.com>
wrote:

> Hi Neil,
>
> On behalf of Maciek I would like to ask you or Bobby to take over this PR.
> In October Maciek will not have enough time to finalize this job.
>
> Cheers,
> Adam Podogrocki
>
> 2017-10-09 13:55 GMT+02:00 Neil Armstrong <notifications@github.com>:
>
> > @MaciejDebski <https://github.com/maciejdebski> Would you want @b0661
> > <https://github.com/b0661> or me to take over this PR (while keeping
> your
> > authorship) if you don't have enough time ?
> > We have at least 2 more F0 SoCs/Boards to submit following this PR
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-
> 335136071>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AP5E9W_K3-
> G1V6SwLBWIKCNqnPORSuZ4ks5sqgmngaJpZM4O2FiW>
> > .
> >
>
>
>
> --
>
> Adam Podogrocki
>
> *Senior Software Engineer* of *rndity;*
>
> +48 607 236 982 <+48%20607%20236%20982> <+48+607+236+982> rndity.com
>
> [image: facebook] <http://facebook.com/rndity> [image: twitter]
> <http://twitter.com/RnDity_com> [image: linkedin]
> <https://www.linkedin.com/company/rndity> [image: github]
> <https://github.com/RnDity>
>
> [image: rndity]
>
> Niniejsza wiadomość wraz z załącznikami może zawierać chronione prawem
> lub poufne informacje i została wysłana wyłącznie do wiadomości i
> użytku osób, do których została zaadresowana. Jeśli wiadomość została
> otrzymana przypadkowo zabrania się jej kopiowania lub rozsyłania do
> osób trzecich. W takim przypadku uprasza się o natychmiastowe
> zniszczenie wiadomości oraz poinformowanie nadawcy o zaistniałej
> sytuacji za pomocą wiadomości zwrotnej. Dziękujemy.
>
> This message, including any attachments hereto, may contain privileged
> or confidential information and is sent solely for the attention and
> use of the intended addressee(s). If you are not an intended
> addressee, you may neither use this message nor copy or deliver it to
> anyone. In such case, you should immediately destroy this message and
> kindly notify the sender by reply email. Thank you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-335139975>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARtr5HAbwtjUXMPwc55RD2JRENkz1U6Tks5sqg30gaJpZM4O2FiW>
> .
>

apodogrocki commented:
Hi Erwan,

I was convinced that there is no more work for me regarding mentioned PR
(iwdg). Patch for CubeMX is required, as it was discussed in separate PR,
but it doesn't cause the iwdg works incorrectly. So if you have nay
suggestions what I could add more, please let me know.

Cheers,
Adam

2017-10-11 10:53 GMT+02:00 Erwan Gouriou <notifications@github.com>:

> Hi Adam,
>
> Should we consider this is same situation for you and
> https://github.com/zephyrproject-rtos/zephyr/pull/556 ?
> ("iwdg: make independent watchdog driver compliant with STM32Cube LL API")
> (keeping your credit of course).
> There is less demand on this one than on STM32F0, but it would be nice we
> don't loose the benefit of your work.
> Or you already planned to resume it later on?
>
> Cheers
> Erwan
>
> On 9 October 2017 at 14:13, Adam Podogrocki <notifications@github.com>
> wrote:
>
> > Hi Neil,
> >
> > On behalf of Maciek I would like to ask you or Bobby to take over this
> PR.
> > In October Maciek will not have enough time to finalize this job.
> >
> > Cheers,
> > Adam Podogrocki
> >
> > 2017-10-09 13:55 GMT+02:00 Neil Armstrong <notifications@github.com>:
> >
> > > @MaciejDebski <https://github.com/maciejdebski> Would you want @b0661
> > > <https://github.com/b0661> or me to take over this PR (while keeping
> > your
> > > authorship) if you don't have enough time ?
> > > We have at least 2 more F0 SoCs/Boards to submit following this PR
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-
> > 335136071>,
> > > or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/AP5E9W_K3-
> > G1V6SwLBWIKCNqnPORSuZ4ks5sqgmngaJpZM4O2FiW>
> > > .
> > >
> >
> >
> >
> > --
> >
> > Adam Podogrocki
> >
> > *Senior Software Engineer* of *rndity;*
> >
> > +48 607 236 982 <+48%20607%20236%20982> <+48%20607%20236%20982>
> <+48+607+236+982> rndity.com
> >
> > [image: facebook] <http://facebook.com/rndity> [image: twitter]
> > <http://twitter.com/RnDity_com> [image: linkedin]
> > <https://www.linkedin.com/company/rndity> [image: github]
> > <https://github.com/RnDity>
> >
> > [image: rndity]
> >
> > Niniejsza wiadomość wraz z załącznikami może zawierać chronione prawem
> > lub poufne informacje i została wysłana wyłącznie do wiadomości i
> > użytku osób, do których została zaadresowana. Jeśli wiadomość została
> > otrzymana przypadkowo zabrania się jej kopiowania lub rozsyłania do
> > osób trzecich. W takim przypadku uprasza się o natychmiastowe
> > zniszczenie wiadomości oraz poinformowanie nadawcy o zaistniałej
> > sytuacji za pomocą wiadomości zwrotnej. Dziękujemy.
> >
> > This message, including any attachments hereto, may contain privileged
> > or confidential information and is sent solely for the attention and
> > use of the intended addressee(s). If you are not an intended
> > addressee, you may neither use this message nor copy or deliver it to
> > anyone. In such case, you should immediately destroy this message and
> > kindly notify the sender by reply email. Thank you.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-
> 335139975>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> ARtr5HAbwtjUXMPwc55RD2JRENkz1U6Tks5sqg30gaJpZM4O2FiW>
> > .
>
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-335742354>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP5E9WoYegFMoElFBr508RyANRE_ih4aks5srIIGgaJpZM4O2FiW>
> .
>



-- 

Adam Podogrocki

*Senior Software Engineer* of *rndity;*

+48 607 236 982 <+48+607+236+982>  rndity.com

[image: facebook] <http://facebook.com/rndity> [image: twitter]
<http://twitter.com/RnDity_com> [image: linkedin]
<https://www.linkedin.com/company/rndity> [image: github]
<https://github.com/RnDity>

[image: rndity]

Niniejsza wiadomość wraz z załącznikami może zawierać chronione prawem
lub poufne informacje i została wysłana wyłącznie do wiadomości i
użytku osób, do których została zaadresowana. Jeśli wiadomość została
otrzymana przypadkowo zabrania się jej kopiowania lub rozsyłania do
osób trzecich. W takim przypadku uprasza się o natychmiastowe
zniszczenie wiadomości oraz poinformowanie nadawcy o zaistniałej
sytuacji za pomocą wiadomości zwrotnej. Dziękujemy.

This message, including any attachments hereto, may contain privileged
or confidential information and is sent solely for the attention and
use of the intended addressee(s). If you are not an intended
addressee, you may neither use this message nor copy or deliver it to
anyone. In such case, you should immediately destroy this message and
kindly notify the sender by reply email. Thank you.

erwango commented:
Yeah, I also considered it was ok.
So, we just miss LSI_VALUE fix to be provided aside and we're good.
And a PR from Yannis just came in. So let's try to get it merged then.

Can you rebase the patchset first?

On 11 October 2017 at 12:16, Adam Podogrocki <notifications@github.com>
wrote:

> Hi Erwan,
>
> I was convinced that there is no more work for me regarding mentioned PR
> (iwdg). Patch for CubeMX is required, as it was discussed in separate PR,
> but it doesn't cause the iwdg works incorrectly. So if you have nay
> suggestions what I could add more, please let me know.
>
> Cheers,
> Adam
>
>
> 2017-10-11 10:53 GMT+02:00 Erwan Gouriou <notifications@github.com>:
>
> > Hi Adam,
> >
> > Should we consider this is same situation for you and
> > https://github.com/zephyrproject-rtos/zephyr/pull/556 ?
> > ("iwdg: make independent watchdog driver compliant with STM32Cube LL
> API")
> > (keeping your credit of course).
> > There is less demand on this one than on STM32F0, but it would be nice we
> > don't loose the benefit of your work.
> > Or you already planned to resume it later on?
> >
> > Cheers
> > Erwan
> >
> > On 9 October 2017 at 14:13, Adam Podogrocki <notifications@github.com>
> > wrote:
> >
> > > Hi Neil,
> > >
> > > On behalf of Maciek I would like to ask you or Bobby to take over this
> > PR.
> > > In October Maciek will not have enough time to finalize this job.
> > >
> > > Cheers,
> > > Adam Podogrocki
> > >
> > > 2017-10-09 13:55 GMT+02:00 Neil Armstrong <notifications@github.com>:
> > >
> > > > @MaciejDebski <https://github.com/maciejdebski> Would you want
> @b0661
> > > > <https://github.com/b0661> or me to take over this PR (while keeping
> > > your
> > > > authorship) if you don't have enough time ?
> > > > We have at least 2 more F0 SoCs/Boards to submit following this PR
> > > >
> > > > —
> > > > You are receiving this because you were mentioned.
> > > > Reply to this email directly, view it on GitHub
> > > > <https://github.com/zephyrproject-rtos/zephyr/
> pull/1103#issuecomment-
> > > 335136071>,
> > > > or mute the thread
> > > > <https://github.com/notifications/unsubscribe-auth/AP5E9W_K3-
> > > G1V6SwLBWIKCNqnPORSuZ4ks5sqgmngaJpZM4O2FiW>
> > > > .
> > > >
> > >
> > >
> > >
> > > --
> > >
> > > Adam Podogrocki
> > >
> > > *Senior Software Engineer* of *rndity;*
> > >
> > > +48 607 236 982 <+48%20607%20236%20982> <+48%20607%20236%20982>
> <+48%20607%20236%20982>
> > <+48+607+236+982> rndity.com
> > >
> > > [image: facebook] <http://facebook.com/rndity> [image: twitter]
> > > <http://twitter.com/RnDity_com> [image: linkedin]
> > > <https://www.linkedin.com/company/rndity> [image: github]
> > > <https://github.com/RnDity>
> > >
> > > [image: rndity]
> > >
> > > Niniejsza wiadomość wraz z załącznikami może zawierać chronione prawem
> > > lub poufne informacje i została wysłana wyłącznie do wiadomości i
> > > użytku osób, do których została zaadresowana. Jeśli wiadomość została
> > > otrzymana przypadkowo zabrania się jej kopiowania lub rozsyłania do
> > > osób trzecich. W takim przypadku uprasza się o natychmiastowe
> > > zniszczenie wiadomości oraz poinformowanie nadawcy o zaistniałej
> > > sytuacji za pomocą wiadomości zwrotnej. Dziękujemy.
> > >
> > > This message, including any attachments hereto, may contain privileged
> > > or confidential information and is sent solely for the attention and
> > > use of the intended addressee(s). If you are not an intended
> > > addressee, you may neither use this message nor copy or deliver it to
> > > anyone. In such case, you should immediately destroy this message and
> > > kindly notify the sender by reply email. Thank you.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-
> > 335139975>,
> > > or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > ARtr5HAbwtjUXMPwc55RD2JRENkz1U6Tks5sqg30gaJpZM4O2FiW>
> > > .
> >
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-
> 335742354>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> AP5E9WoYegFMoElFBr508RyANRE_ih4aks5srIIGgaJpZM4O2FiW>
> > .
> >
>
>
>
> --
>
> Adam Podogrocki
>
> *Senior Software Engineer* of *rndity;*
>
> +48 607 236 982 <+48%20607%20236%20982> <+48+607+236+982> rndity.com
>
> [image: facebook] <http://facebook.com/rndity> [image: twitter]
> <http://twitter.com/RnDity_com> [image: linkedin]
> <https://www.linkedin.com/company/rndity> [image: github]
> <https://github.com/RnDity>
>
> [image: rndity]
>
> Niniejsza wiadomość wraz z załącznikami może zawierać chronione prawem
> lub poufne informacje i została wysłana wyłącznie do wiadomości i
> użytku osób, do których została zaadresowana. Jeśli wiadomość została
> otrzymana przypadkowo zabrania się jej kopiowania lub rozsyłania do
> osób trzecich. W takim przypadku uprasza się o natychmiastowe
> zniszczenie wiadomości oraz poinformowanie nadawcy o zaistniałej
> sytuacji za pomocą wiadomości zwrotnej. Dziękujemy.
>
> This message, including any attachments hereto, may contain privileged
> or confidential information and is sent solely for the attention and
> use of the intended addressee(s). If you are not an intended
> addressee, you may neither use this message nor copy or deliver it to
> anyone. In such case, you should immediately destroy this message and
> kindly notify the sender by reply email. Thank you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/1103#issuecomment-335764096>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARtr5KrBzLnX0Y8Ler-S1NIyT_Tc-y-rks5srJWKgaJpZM4O2FiW>
> .
>

superna9999 commented:
@galak can you close this PR ? I sent a new one on #4260
